### PR TITLE
Cozette: Init at 1.5.1

### LINKED ISSUE #80267: Package request: Cozette font
**Project description**
A bitmap font optimized for coziness.

**Metadata**

* homepage URL: https://github.com/slavfox/Cozette
* source URL: https://github.com/slavfox/Cozette
* license: MIT
* platforms: Linux

### PR DESCRIPTION
**Pull request by brettlyons**
###### Motivation for this change

Package request: Cozette font #80267

Fixes #80267

cc @rnhmjoj @soundtoxin

### CODE REVIEW

File: pkgs/data/fonts/cozette/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    cp -a *.ttf $out/share/fonts/truetype/
+    mkfontdir "$out/share/fonts/misc"
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A bitmap programming font optimized for coziness.";
+    homepage = "https://github.com/slavfox/cozette";
+    license = licenses.mit;
+    platforms = platforms.all;
+    maintainers = with maintainers; [ ];
```
Comment (cole-h) [inline pkgs/data/fonts/cozette/default.nix]: This (`maintainers`) should be set. See https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes, the last bullet. If you add yourself as the maintainer, be sure to do so in a separate commit, please.

File: pkgs/data/fonts/cozette/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  unpackCmd = ''
+    otName=$(basename $(stripHash $curSrc))
+    cp $curSrc ./$otName
+  '';
+
+  installPhase = ''
+    mkdir -p $out/share/fonts/misc
+    mkdir -p $out/share/fonts/truetype
+    cp -a *.otf $out/share/fonts/misc/
```
Comment (cole-h) [inline pkgs/data/fonts/cozette/default.nix]: I believe `.otf` fonts should go to `$out/share/fonts/opentype` -- this is why I think that: https://github.com/NixOS/nixpkgs/blob/92442113a39bd1c4d6b9923df71c95a70e32de8f/pkgs/data/fonts/unscii/default.nix#L32

File: pkgs/data/fonts/cozette/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
+{ stdenv, fetchurl, mkfontscale }:
+
+stdenv.mkDerivation rec {
+  pname = "Cozette";
+  version = "1.5.1";
+  srcs = [
+    (fetchurl {
+      url = "https://github.com/slavfox/Cozette/releases/download/v." + version
+        + "/cozette.otb";
```
Comment (cole-h) [inline pkgs/data/fonts/cozette/default.nix]: I believe the more idiomatic, Nix-y, and nicer-looking way to do this is:

```nix
url = "https://github.com/slavfox/Cozette/releases/download/v.${version}/cozette.otb";
```

**Review by cole-h — CHANGES_REQUESTED** [review]
I was looking into packaging this myself, but wanted to try building it from source. As you can probably tell, that didn't go according to plan, so thanks for picking this up!

**Comment by rnhmjoj** [comment]
> I was looking into packaging this myself, but wanted to try building it from source. As you can probably tell, that didn't go according to plan, so thanks for picking this up!

I think it's fine not to build from source for a font. Given the package is simply copying files it could even be a fixed output derivation.

File: pkgs/data/fonts/cozette/default.nix (hunk 1 above)
Comment (brettlyons) [reply pkgs/data/fonts/cozette/default.nix]: Done :)

File: pkgs/data/fonts/cozette/default.nix (hunk 2 above)
Comment (brettlyons) [reply pkgs/data/fonts/cozette/default.nix]: Okay.  Done :)

File: pkgs/data/fonts/cozette/default.nix (hunk 1 above)
Comment (cole-h) [reply pkgs/data/fonts/cozette/default.nix]: Just a slight nitpick: commit title for adding a maintainer should start with `maintainers:`, since that's the file being modified, not the module you're becoming the maintainer for ;)

File: pkgs/data/fonts/cozette/default.nix [hunk 4]
```diff
@@ -0,0 +1,59 @@
+{ stdenv, fetchurl, mkfontscale }:
+
+let
+  version = "1.5.1";
+  releaseUrl =
+    "https://github.com/slavfox/Cozette/releases/download/v.${version}";
+in stdenv.mkDerivation rec {
```
Comment (cole-h) [inline pkgs/data/fonts/cozette/default.nix]: Looks like you clobbered the `pname` and `version` attributes. Please add them back. Since you `let`-bind `version`, I believe you can just use an `inherit version;` instead of `version = version;`.

File: pkgs/data/fonts/cozette/default.nix [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+    mkdir -p $out/share/fonts/misc
+    cp -a *.otb $out/share/fonts/misc/
+
+    mkfontdir "$out/share/fonts/misc"
+    mkfontscale "$out/share/fonts/truetype"
+    mkfontscale "$out/share/fonts/opentype"
+  '';
+
+  outputs = [ "out" ];
```
Comment (cole-h) [inline pkgs/data/fonts/cozette/default.nix]: Not that this really matters, but if you don't specify any `outputs`, it defaults to `out`.

File: pkgs/data/fonts/cozette/default.nix [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+  meta = with stdenv.lib; {
+    description = "A bitmap programming font optimized for coziness.";
+    homepage = "https://github.com/slavfox/cozette";
+    license = licenses.mit;
+    platforms = platforms.all;
+    maintainers = with maintainers; [ brettlyons ];
```
Comment (cole-h) [inline pkgs/data/fonts/cozette/default.nix]: Just to ensure my edit above didn't get lost to the void:

You should add yourself to the `maintainers` list in the same commit you init the package, and adding yourself to the `maintainers` file should its own commit, touching only that file (just to make it easier to `git blame`), with a message of `maintainers: add brettlyons` (because "myself" doesn't mean anything to anybody but you).

**Review by cole-h — CHANGES_REQUESTED** [review]

File: pkgs/data/fonts/cozette/default.nix (hunk 4 above)
Comment (brettlyons) [reply pkgs/data/fonts/cozette/default.nix]: Okay.

File: pkgs/data/fonts/cozette/default.nix (hunk 6 above)
Comment (brettlyons) [reply pkgs/data/fonts/cozette/default.nix]: Okay.

File: pkgs/data/fonts/cozette/default.nix [hunk 7]
```diff
@@ -0,0 +1,61 @@
+{ stdenv, fetchurl, mkfontscale }:
+
+let
+  version = "1.5.1";
+  releaseUrl =
+    "https://github.com/slavfox/Cozette/releases/download/v.${version}";
+in stdenv.mkDerivation rec {
+  inherit version;
+  pname = "Cozette";
```
Comment (cole-h) [inline pkgs/data/fonts/cozette/default.nix]:
```suggestion
  pname = "Cozette";
  inherit version;

```

**Review by cole-h — COMMENTED** [review]
My final nitpick: most packages have their `pname` followed by `version`, and add a newline before `src{,s}`. Certainly not a blocking request, but it Just Looks Nicer™.

Thanks for bearing with my super nitpick-y requests!

File: pkgs/data/fonts/cozette/default.nix (hunk 7 above)
Comment (brettlyons) [reply pkgs/data/fonts/cozette/default.nix]: Cool, right on.  Makes sense.  I have another change to add as well.

Comment (brettlyons) [reply pkgs/data/fonts/cozette/default.nix]: That'll do.

**Review by cole-h — APPROVED** [review]
LGTM! Thanks again for dealing with my various nitpicks :)

```
[5 built, 19 copied (13.0 MiB), 3.6 MiB DL]
https://github.com/NixOS/nixpkgs/pull/80913
1 package built:
cozette
```

File: pkgs/data/fonts/cozette/default.nix [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+      sha256 = "1q4ml8shv9lmyc6bwhffwvbvl92s73j7xkb0rkqvci4f0zbz7mcy";
+    }
+  ];
+
+  nativeBuildInputs = [ mkfontscale ];
+
+  sourceRoot = "./";
+
+  unpackCmd = ''
+    otName=$(basename $(stripHash $curSrc))
```
Comment (rnhmjoj) [inline pkgs/data/fonts/cozette/default.nix]: The `basename` is redundant, also the variable must be quoted.
```suggestion
    otName=$(stripHash "$curSrc")
```

File: pkgs/data/fonts/cozette/default.nix [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    cp $curSrc ./$otName
+  '';
+
+  installPhase = ''
+
+    mkdir -p $out/share/fonts/opentype
+    cp -a *.otf $out/share/fonts/opentype/
```
Comment (rnhmjoj) [inline pkgs/data/fonts/cozette/default.nix]: You can drop the `mkdir`  and do `install -D -m 644 *.otf -t "$out/share/fonts/opentype"`

**Review by rnhmjoj — CHANGES_REQUESTED** [review]

File: pkgs/data/fonts/cozette/default.nix (hunk 9 above)
Comment (brettlyons) [reply pkgs/data/fonts/cozette/default.nix]: Okay.  Done.

**Review by rnhmjoj — APPROVED** [review]
Ok, looks all good now. Thank you.
